File: creational/python/factory_method/CarCreator.py
from python.factory_method.Audi     import Audi
from python.factory_method.Mercedes import Mercedes
from python.factory_method.Bmw      import Bmw
from python.factory_method.Car      import car_interface
class car_creator:
    def create_car(self, brand) -> car_interface:
        if(brand == "Audi"):
            return Audi()
        elif(brand == "Bmw"):
            return Bmw()
        else:
            return Mercedes()

File: creational/java/factory_method/src/CarCreator.java
public class CarCreator
{
    public Car createCar(String brand)
    {
        if (brand.equals("Audi"))
        {
            return new Audi();
        }
        else if (brand.equals("Bmw"))
        {
            return new Bmw();
        }
        else
        {
            return new Mercedes();
        }
    }
}
